### ISSUE TITLE
[BUG] The result is inconsistent after serializing and deserializing ChatRequestUserMessage

### ISSUE THREAD

**Issue by fairjm** [Azure/azure-sdk-for-java#40481]
**Describe the bug**
After serializing ChatRequestUserMessage and then deserializing it, the structure of the result returned is inconsistent with the original structure.  
It seems that BinaryData directly converted the List into a string structure.

***Exception or Stack Trace***

**To Reproduce**
run the following code:

***Code Snippet***
```java
        final ChatRequestUserMessage user = new ChatRequestUserMessage(List.of(
                new ChatMessageTextContentItem("text"),
                new ChatMessageImageContentItem(new ChatMessageImageUrl("testImage"))
        ));
        // log.info("user:{}", user);
        final String content = BinaryData.fromObject(user).toString();
        log.info("json content:{}", content);
        final ChatRequestUserMessage converted = BinaryData.fromString(content).toObject(ChatRequestUserMessage.class);
        // log.info("converted:{}", converted);
```

**Expected behavior**
converted should have the same structure as user.

**Screenshots**
the user:
![image](https://github.com/Azure/azure-sdk-for-java/assets/4281721/d99c0a3b-a29f-46a4-8cf8-161a26160eb0)

the converted:
![image](https://github.com/Azure/azure-sdk-for-java/assets/4281721/cc33d00d-21e5-41b8-839e-b52cb7a1ddd1)

**Setup (please complete the following information):**
 - OS: windows 11
 - IDE: IntelliJ
 - Library/Libraries: azure-ai-openai 1.0.0-beta.8 
 - Java version: 17

**Comment by fairjm** [comment]
I found a relatively simple temporary solution - perform an additional conversion on the content of ChatRequestUserMessage, but this method would also bring some strange issues if the text content itself is an array.   
Not good but it works.
```java
final BinaryData b = converted.getContent();
try {
    final List<ChatMessageContentItem> items = b.toObject(new TypeReference<>() {
    });
    converted = new ChatRequestUserMessage(items);
} catch (Exception ignored) {}
```

**Comment by mssfang** [comment]
@fairjm Thank you for reporting this bug. I have a fix in linked PR #40687. It should fix the issue.

**Comment by fairjm** [comment]
sorry, but I found another issue.
I update to beta.10.
The deserialized object is :
![image](https://github.com/user-attachments/assets/a9cd397a-13eb-413f-bbdf-08f909329d83)
You can see `stringContent` and `chatMessageContentItems` are both null.
And the toJson method in `ChatRequestUserMessage`:
```java
    @Override
    public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
        jsonWriter.writeStartObject();
        if (stringContent != null) {
            jsonWriter.writeStringField("content", stringContent);
        } else if (chatMessageContentItems != null) {
            jsonWriter.writeArrayField("content", chatMessageContentItems, JsonWriter::writeJson);
        }
        jsonWriter.writeStringField("role", this.role == null ? null : this.role.toString());
        jsonWriter.writeStringField("name", this.name);
        return jsonWriter.writeEndObject();
    }
```

it means it will return no content if toJson:
![image](https://github.com/user-attachments/assets/27c59283-a3fd-4d11-bfd0-4d26e95c1b94)

Send request to openAI will fail in OpenAIClient:
```java
    @Generated
    @ServiceMethod(returns = ReturnType.SINGLE)
    public ChatCompletions getChatCompletions(String deploymentOrModelName,
        ChatCompletionsOptions chatCompletionsOptions) {
        // Generated convenience method for getChatCompletionsWithResponse
        RequestOptions requestOptions = new RequestOptions();
        return getChatCompletionsWithResponse(deploymentOrModelName, BinaryData.fromObject(chatCompletionsOptions),
            requestOptions).getValue().toObject(ChatCompletions.class);
    }
```

I think you'd better check it out. @mssfang
